thread_pool.c

```diff
@@ -15,8 +15,7 @@ pthread_t thread_pool[THREAD_POOL_SIZE];
 void *
 thread_start_routine(void *arg) {
     while (true) {
-        task_t *task = task_queue_retrieve(&task_queue);
-        task->action(task->argument);
+        task_execute(task_queue_retrieve(&task_queue));
```

Comment:
As this does not yet use the new `task` module, this function is not available. This change should be moved to the next commit.

Reply:
Changed accordingly.